src/crimson/os/alienstore/alien_store.cc

```diff
@@ -301,11 +301,12 @@ AlienStore::get_attr(CollectionRef ch,
 {
   logger().debug("{}", __func__);
   assert(tp);
-  return seastar::do_with(ceph::bufferlist{}, [=] (auto &value) {
-    return tp->submit(ch->get_cid().hash_to_shard(tp->size()), [=, &value] {
+  return seastar::do_with(ceph::bufferlist{}, std::string{name},
+                          [=] (auto &value, const auto& name) {
+    return tp->submit(ch->get_cid().hash_to_shard(tp->size()), [=, &value, &name] {
+      static_assert(std::is_same_v<decltype(name), const std::string&>);
```

Comment:
just not sure if we need this line. for two reasons

1. this piece of code is not templated. so we always have a `const string&` passed to the continuation in `do_with()`.
2. the code path which determines the type of `name` is very close to where it is consumed. and it is straightforward that it is a `const string&`

but i understand that you wanted to highlight that the lifecycle of `name` is extended. but practically, the same applies to all variables created with`do_with()`, shall we `static_assert()` them as well once we are bitten by a use-after-free issue ?

Reply:
> i understand that you wanted to highlight that the lifecycle of `name` is extended. but practically, the same applies to all variables created with `do_with()`, shall we `static_assert()` them as well once we are bitten by a use-after-free issue ?

Yes, the intention is to emphasize the life-time extension which happens in a quite unusual way: `name` basically changes its type to prolong data of the `string_view` as well. This type alternation is the centerpiece. Perhaps a comment would be less disturbing. How about that?

Reply:
@rzarzynski yeah, a comment would be great! thank you.